app/Models/MatchingStrategy.php

```diff
@@ -39,27 +45,24 @@ public function matchable(): MorphTo
 
     public function hasDisabilityTypes(): bool
     {
-        return $this->criteria()->where('criteriable_type', DisabilityType::class)->count() > 0;
+        return $this->identities()->where('cluster', IdentityCluster::DisabilityAndDeaf)->count() > 0;
     }
 
-    public function hasDisabilityType(DisabilityType $disabilityType): bool
+    public function hasDisabilityType(Identity $disabilityType): bool
     {
-        return $this->criteria()->where('criteriable_type', DisabilityType::class)->with('criteriable')->get()->pluck('criteriable')->contains($disabilityType);
+        return $this->identities->contains($disabilityType);
     }
 
-    public function hasCriterion(string $model, mixed $criteriable): bool
+    public function hasIdentity(Identity $identity): bool
     {
-        return $this->criteria()->where('criteriable_type', $model)->with('criteriable')->get()->pluck('criteriable')->contains($criteriable);
+        return $this->identities->contains($identity);
     }
 
-    public function hasCriteria(string $model, array $criteriables): bool
+    public function hasIdentities(array $identities): bool
     {
-        $items = $model::whereIn(
-            'id',
-            Arr::map($criteriables, fn ($criteriable) => $criteriable->id)
-        )->get();
+        $identities = collect($identities);
 
-        return $this->criteria()->where('criteriable_type', $model)->with('criteriable')->get()->pluck('criteriable')->intersect($items)->count() === $items->count();
+        return $this->identities->intersect($identities)->count() === $identities->count();
```

Comment:
can you use the [`Arr::has()`](https://laravel.com/docs/9.x/helpers#method-array-has) helper method for this?

Reply:
I'm not sure that would work as `$this->identities` is a collection. Were you thinking that it would make sense to turn it into an array and then use `Arr:has()`? I'm also not sure about the dot notation approach. Let's discuss.

Reply:
@jobara I'm handling this by making the method accept an array of IDs instead of an array of Identities. Then I can use `Arr::has` without issue.